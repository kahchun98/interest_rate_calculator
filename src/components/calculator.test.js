import renderer from 'react-test-renderer';
import Calculator from './calculator';

test('Calculator snapshot', () => {
  const component = renderer.create(<Calculator />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
