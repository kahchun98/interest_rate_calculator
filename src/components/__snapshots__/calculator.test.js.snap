// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot 1`] = `
<div
  className="calculatorWrapper"
>
  <p>
    <label>
      Starting Amount: 
      <input
        id="principalAmount"
        name="principalAmount"
        onChange={[Function]}
        step="100"
        type="text"
        value={0}
      />
    </label>
  </p>
  <label>
    Interest Rate: 
    <input
      id="interestRate"
      max="15"
      min="0"
      name="interestRate"
      onChange={[Function]}
      step="0.01"
      type="range"
      value={0}
    />
    0
    % p.a.
  </label>
  <p>
    <label>
      Year: 
      <input
        id="termYear"
        max="5"
        min="0"
        name="termYear"
        onChange={[Function]}
        type="range"
        value={0}
      />
       
      0
    </label>
  </p>
  <label>
    Month: 
    <input
      id="termYear"
      max="12"
      min="0"
      name="termMomth"
      onChange={[Function]}
      type="range"
      value={0}
    />
     
    0
  </label>
  <div>
    <button
      onClick={[Function]}
    >
      Calculate at maturity interest
    </button>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Calculate monthly interest
    </button>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Calculate quarterly interest
    </button>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Calculate yearly interest
    </button>
  </div>
  <p>
    Total balance: 
    0
  </p>
  Total interest earned: 
  0
</div>
`;
